### ISSUE TITLE
GATED research: pairwise salience learning for Breakdown and Changes

### ISSUE THREAD

**Issue by gr-rr** [ListenCloser/listencloser#1307]
Program owner: #1303.
Research synthesis: #1297.
Relevant product owners: #588 / Breakdown, #848 (Changes), #812 (Similar Moments), #1161 (Inspector hierarchy where still relevant).

## Status

**GATED SECOND WAVE — do not start model training until a small trustworthy preference dataset exists.**

This issue formalizes a plausible proprietary ListenCloser ML layer without pretending the required labels already exist.

## Question

> **Can a tiny learned ranker choose which already-admitted musical findings are most useful to show a musician better than the current deterministic presentation policy?**

This is a ranking problem, not a truth-estimation problem. The model must never change whether a finding is musically admitted/true; it only ranks true/qualified candidates.

## Why this is product-specific

There is no standard MIR benchmark for:

> “Of the true things we know about this exact selected passage, which 3–5 are worth showing this musician first?”

Current LC ranking already encodes product policy such as selection overlap, finding-kind priors, maturity penalties, evidence breadth and dedupe. That is a strong baseline and a plausible source of features, not a reason to train a large model.

## Gate to start

Do not begin model fitting until all are true:

- [ ] current deterministic ranking baseline is reproducible on a frozen candidate set;
- [ ] at least ~100 high-quality pairwise judgments exist across multiple Works/selections, with a target of ~200–500 if collection is cheap;
- [ ] judgments compare candidates drawn from the **same context** rather than unrelated Works;
- [ ] position/UI exposure is not silently used as the label;
- [ ] train/dev/test split is by Work or at least by selection context to avoid leakage.

If these labels do not materialize naturally, leave this issue gated rather than inventing proxy click labels.

## Data shape

A training row should look approximately like:

```text
work_id / source hash
selection start/end
candidate A id + features
candidate B id + features
preferred: A | B | tie/unclear
label source / annotator class
```

Candidate text itself may be stored for audit, but the first model should prefer structured finding/evidence features over embedding free-form prose.

## Initial feature set

Use small, inspectable features already available in product truth/presentation logic where possible:

- selection overlap / distance;
- finding kind/type;
- temporal scope;
- evidence breadth;
- measured magnitude/effect size where legitimate;
- recurrence/novelty indicators when available;
- maturity/qualification state;
- whether finding adds information not already represented by another candidate;
- source/method confidence only when it is a calibrated/task-valid quantity, not generic model confidence.

Do not include user identity/personalization in v0.

## Models

Compare only small models first:

```text
A. current deterministic ranking
B. Bradley-Terry / logistic pairwise ranker
C. gradient-boosted tree OR tiny MLP if justified by feature interactions
```

No LLM fine-tuning, RL, transformer ranker, or bandit in this issue.

## Metrics

Primary:
- held-out pairwise preference accuracy.

Secondary:
- NDCG@3 / NDCG@5 on contexts with enough ranked candidates;
- calibration of pairwise preference probability if useful;
- breakdown by finding kind and Work/domain;
- disagreement/unclear-label rate.

A useful model must generalize to held-out Works/selections, not just memorize annotator/context priors.

## Compute budget

Expected CPU-only.

```text
GPU-hours = 0 preferred
cash cost = $0
```

scikit-learn / XGBoost-like tooling is sufficient if already acceptable; prefer scikit-learn first to avoid dependency growth.

## Success criterion

Proceed only if a tiny learned ranker materially beats the deterministic baseline on held-out contexts **without weakening truth/qualification boundaries**.

A strong outcome would be:

- consistent pairwise lift across held-out Works;
- useful ranking lift at top-3/top-5;
- interpretable feature/error analysis;
- no need for a large model.

## Stop / kill criteria

Stop if:

- pairwise labels are too inconsistent/subjective to support a stable ranking signal;
- the deterministic baseline is already near annotator agreement ceiling;
- performance relies mainly on finding type priors or leakage;
- useful generalization requires user-specific modeling or much larger datasets;
- click/position bias is the only scalable label source.

## Product consequence if successful

A successful model may replace only the **ranking** portion of Breakdown/Changes presentation after a focused product validation. It must consume admitted evidence and preserve source/provenance/qualification.

Potential later work, out of scope here:
- active-learning query selection;
- contextual bandits;
- personalization adapters;
- preference optimization of a generative explanation layer.

## Deliverables

1. Frozen pairwise preference dataset with source/selection-disjoint evaluation.
2. Reproducible deterministic baseline.
3. Small-model comparison.
4. Held-out metrics + annotator agreement/label-noise analysis.
5. Feature importance/error analysis.
6. Explicit disposition:

```text
FAIL — no stable learnable salience beyond current policy
NARROW — small gain but insufficient product justification
CONTINUE — clear held-out ranking lift; open focused product integration owner
```

## Acceptance

- [ ] gate conditions met before training;
- [ ] no click-position proxy silently substitutes for preference labels;
- [ ] deterministic baseline vs small learned ranker compared on held-out contexts;
- [ ] CPU/$0 target maintained;
- [ ] truth/admission logic remains outside the model;
- [ ] disposition written back to #1303.

Refs #1303 #1297 #588 #848 #812 #1161.

**Comment by gr-rr** [comment]
## Program constraint update: no bespoke manual labeling assumed

Current #1303 execution policy assumes we will **not** launch a dedicated human-labeling effort for this pilot program.

Keep this issue gated unless a trustworthy same-context preference dataset appears organically or from a public/existing source. Do not manufacture the missing musician-preference ground truth with an LLM or click-position proxy.

LLM judgments are acceptable only as explicitly weak/train-only supervision or exploratory disagreement diagnostics; held-out success must still be measured against legitimate human/public preference evidence.

Until that data exists, spending compute here is not justified. Refs #1303 #1311.
